Automattic/sensei
# Add sample course installation to the Setup Wizard

Fixes #3499

### Changes proposed in this Pull Request

* Add sample course installation to Setup Wizard.

Observation: We can already review the code, but it won't be merged until we have the Exporter ready to export the demo course from `senseilms.com` and add the correct CSV files. For now, I did a cherry-pick from the sample CSVs commit and I'm using it.

In the client, it was created isolated from the store, because it's a more isolated thing that doesn't impact in other parts of the Setup Wizard.

This PR doesn't handle the user that already installed the sample course. If they return to the Setup Wizard, they can install it again. If we desire something different from that, we can just create an `option` to identify that it already happened and block the button (or remove the block). In my opinion, I think keeping it always active is the best approach, because the user can re-install it if they want to test something in the future.

### Testing instructions

* Go to the Setup Wizard, and navigate until the ready step.
* Click on `Install a sample course`.
* Wait until the installation and make sure you're redirected to the courses page with the course created.

You can also simulate an error, changing the file `assets/setup-wizard/ready/use-sample-course.js`. You can replace the `start-sample` path to `start-sampleX`.


### Screenshot / Video

<img width="562" alt="Screen Shot 2020-08-06 at 15 15 23" src="https://user-images.githubusercontent.com/876340/89569616-5e796a80-d7fb-11ea-8efa-ee1694884790.png">

<img width="548" alt="Screen Shot 2020-08-06 at 15 15 30" src="https://user-images.githubusercontent.com/876340/89569626-60dbc480-d7fb-11ea-84af-186f8aa02b9d.png">

<img width="534" alt="Screen Shot 2020-08-06 at 15 16 01" src="https://user-images.githubusercontent.com/876340/89569637-646f4b80-d7fb-11ea-8f69-3dfc6b4b3597.png">

Notice that this is a flexbox element, and the content is vertically centered aligned. So, with the error, button column grows and it moves a little the content in the text column. We could change the alignment, to be on top, for example, but I don't think it worths it because it's only for the error case that will likely be rare. But if someone has a different opinion about that, we can change it.
